scripts/build.sh

```diff
@@ -20,12 +20,33 @@ elif [[ "$OSTYPE" == "darwin"* ]]; then
     CMAKE_FLAGS+="-DCMAKE_C_COMPILER=${XCODE_CMDLINE_DIR}/usr/bin/clang -DCMAKE_CXX_COMPILER=${XCODE_CMDLINE_DIR}/usr/bin/clang++ -DCMAKE_CXX_FLAGS=-isystem\ /usr/local/include -DCMAKE_EXPORT_COMPILE_COMMANDS=ON"
 fi
 
-CMAKE_BUILD_TYPE="Debug"
-if [[ "$BUILD_RELEASE" == "1" ]]; then
+# Note:
+# CMAKE_BUILD_TYPE="Debug" adds "-O0 -g" flags by default
+# CMAKE_BUILD_TYPE="Release" adds "-O3 -DNDEBUG" by default
+if [[ "$BUILD_DEBUG" == "1" ]]; then
+    CMAKE_BUILD_TYPE="Debug"
+elif [[ "$BUILD_RELEASE" == "1" ]]; then
     CMAKE_BUILD_TYPE="Release"
 elif [[ "$BUILD_PROFILING" == "1" ]]; then
     CMAKE_BUILD_TYPE="Profiling"
 fi
 
+if [ $# -gt 0 ]; then
+    if [ "$1" == "Release" ]; then
+        CMAKE_BUILD_TYPE="Release"
```

Comment:
[optional] Thinking it could be helpful to have a 'help' arg for users using the cli and/or reading the script. This would show the options of Release|Profiling|Debug for `build.sh` if and only if the user enters 'help' or '--help' exactly.
```suggestion
    if [[ "$1" == "help" || "$1" == "--help" ]]; then
```